### PR TITLE
Use the ecef column vectors to calculate the ecef transform if present so that non-rigid transforms are handled.

### PR DESCRIPTION
**Pull request by RBBentley**
The native code PR: https://bentleycs.visualstudio.com/iModelTechnologies/_git/imodel02/pullrequest/162984  calculates an accurate ecef transform at the project center and adds the X and Y Column vectors to the EcefLocation properties so that non rigid transforms are supported.  This code uses the column vectors if present to calculate the ecefTransform.  This allows elimination of the BackgroundMapLocation code that handled this on the front end and necessitated asynchronous initialization.

### CODE REVIEW

**Comment by RBBentley** [comment]
> Are there any changes to native code?

Yes... I'm still working on this.  Meant to make this draft for now.

**Review by pmconne — CHANGES_REQUESTED** [review]
Did you `rush build`? You made [this change](https://github.com/imodeljs/imodeljs/commit/51d72175c26bf50a2286009d3daf04a66bee5617#diff-9119e386525e6a373a9f204aeaed7a94de157eda082840aed6be2b9b5b482455) to display-performance-test-app that I expect will no longer compile, right?

File: core/frontend/src/IModelConnection.ts [hunk 1]
```diff
@@ -554,9 +549,19 @@ export abstract class IModelConnection extends IModel {
     // setEcefLocation is invoked from IModel constructor...
     if (this.tiles)
       this.tiles.onEcefChanged();
+  }
+  public getMapEcefToDb(bimElevationBias: number): Transform {
```
Comment (pmconne) [inline core/frontend/src/IModelConnection.ts:553]: Is this intentionally `@public`?

**Comment by RBBentley** [comment]
> Did you `rush build`? You made [this change](https://github.com/imodeljs/imodeljs/commit/51d72175c26bf50a2286009d3daf04a66bee5617#diff-9119e386525e6a373a9f204aeaed7a94de157eda082840aed6be2b9b5b482455) to display-performance-test-app that I expect will no longer compile, right?

Fixed.

**Review by pmconne — COMMENTED** [review]
The workaround for background map location moved to TestRunner.setupTest in TestRunner.ts; it can be removed now.

**Comment by RBBentley** [comment]
> The workaround for background map location moved to TestRunner.setupTest in TestRunner.ts; it can be removed now.

I found some additional work involved after merging... hold off on review for now.

**Comment by RBBentley** [comment]
> The workaround for background map location moved to TestRunner.setupTest in TestRunner.ts; it can be removed now.

Done -- and ready now for review.

File: core/frontend/src/DisplayStyleState.ts [hunk 2]
```diff
@@ -796,18 +796,18 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
   public getIsBackgroundMapVisible(): boolean {
     return undefined !== this.iModel.ecefLocation && (this.viewFlags.backgroundMap || this.anyMapLayersVisible(false));
   }
-  public get backgroundMapElevationBias(): number {
+  public get backgroundMapElevationBias(): number | undefined{
```
Comment (pmconne) [inline core/frontend/src/DisplayStyleState.ts:799]: This is `@public`, can you document what it's for?

File: core/common/src/IModel.ts [hunk 3]
```diff
@@ -184,10 +187,22 @@ export class EcefLocation implements EcefLocationProps {
...
   /** Optional position on the earth used to establish the ECEF origin and orientation. */
   public readonly cartographicOrigin?: Cartographic;
+  /** Optional, column vectors for ECEF non-rigid ECEF transform  */
+  public readonly xVector?: Vector3d;
+  public readonly yVector?: Vector3d;
 
   /** Get the transform from iModel Spatial coordinates to ECEF from this EcefLocation */
   public getTransform(): Transform {
-    return Transform.createOriginAndMatrix(this.origin, this.orientation.toMatrix3d());
+    let matrix;
```
Comment (pmconne) [inline core/common/src/IModel.ts:196]: This gets called quite frequently from the frontend, right? Did you consider computing the transform once in the constructor?

File: core/frontend/src/IModelConnection.ts [hunk 4]
```diff
@@ -1020,3 +1029,26 @@ export namespace IModelConnection { // eslint-disable-line no-redeclare
     }
   }
 }
+
+export class ProjectAltitudeProvider {
```
Comment (pmconne) [inline core/frontend/src/IModelConnection.ts:1033]: Missing release tag; supposed to be `@public`? If not, need release tag on IModelConnection.altitudeProvider.

**Review by pmconne — CHANGES_REQUESTED** [review]
Minor things.

**Comment by pmconne** [comment]
Please lint and extract-api before merging - CI job won't do it for you.

**Comment by RBBentley** [comment]
> Please lint and extract-api before merging - CI job won't do it for you.

Done.

File: core/frontend/src/IModelConnection.ts (hunk 4 above)
Comment (RBBentley) [reply core/frontend/src/IModelConnection.ts:1036]: Fixed.

File: core/common/src/IModel.ts (hunk 3 above)
Comment (RBBentley) [reply core/common/src/IModel.ts]: Done.

File: core/frontend/src/DisplayStyleState.ts (hunk 2 above)
Comment (RBBentley) [reply core/frontend/src/DisplayStyleState.ts:800]: Moved to internal.

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Pull request merged**